data-prepper-plugins/s3-source/src/integrationTest/java/com/amazon/dataprepper/plugins/source/S3ObjectWorkerIT.java

```diff
@@ -69,7 +73,9 @@ void setUp() {
 
         pluginMetrics = mock(PluginMetrics.class);
         final Counter counter = mock(Counter.class);
-        when(pluginMetrics.counter(S3_OBJECTS_FAILED_METRIC_NAME)).thenReturn(counter);
+        final Timer timer = new NoopTimer(new Meter.Id("test", Tags.empty(), null, null, Meter.Type.TIMER));
```

Comment:
Why can't we just mock the `Timer` here?

Reply:
We can. But, since this is an integration test and the whole logic of this class is related to the `Timer`, I preferred using the `NoopTimer` to get some real behavior from Micrometer.